app/controllers/landing_page_controller.rb

```diff
@@ -36,7 +42,7 @@ def index
       if cache_meta.nil?
         cache_hit = false
         content = build_html(
-          community_id: cid,
+          community: community,
```

Comment:
For API key your suggestion is easy to do. However this builds on top of Analytics and the analytics partials assume they have access to `@current_community`. So to make those work it would require mode work then.